apis/r/R/SOMASparseNDArray.R

```diff
@@ -74,7 +78,76 @@ SOMASparseNDArray <- R6::R6Class(
         j = bit64::as.integer64(values@j),
         x = values@x
       )
-      colnames(coo) <- c(self$dimnames(), self$attrnames())
+      dnames <- self$dimnames()
+      colnames(coo) <- c(dnames, self$attrnames())
+      ranges <- sapply(
+        X = dnames,
+        FUN = function(x) {
+          return(range(coo[[x]]))
+        },
+        simplify = FALSE,
+        USE.NAMES = TRUE
+      )
+      bbox <- bbox %||% ranges
+      if (is.null(names(bbox))) {
+        names(bbox) <- dnames
+      }
+      if (!is_named(bbox, allow_empty = FALSE)) {
+        idx <- which(!nzchar(names(bbox)))
```

Comment:
R jedi know what this means. the rest of us don't. and yes we can fire up an R session and do `?nzchar` in order to read the code. but please meet the rest of us halfway at least and put in some comments about what each bit accomplishes. you will be doing the world a kindness and a service.

Reply:
Added in 5ef5a2d